src/main/java/seedu/duke/chatbot/Parser.java

```diff
@@ -0,0 +1,227 @@
+            throw new NoValidTaskIndexException();
+        }
+        int indexToUnmark = Integer
+                .parseInt(command
+                        .substring(indexAfterCommand)) - 1;
+        return indexToUnmark;
+    }
+
+    /**
+     * Parses a command that seeks to delete a task.
+     * @param command is a string with a command keyword "delete" from user
+     * @return the index of the {@link Task} to delete in {@link seedu.duke.task.TaskList}
+     * @throws DukeException if an invalid task number is given from user
+     */
+    int parseForDelete(String command) throws DukeException {
+        int indexTaskName = 7;
+        //command is given as "delete <taskIndex>"
+        if (command.length() <= 7) { //e.g. "delete " vs "delete 1" (correct)
+            throw new NoValidTaskIndexException();
+        }
+        //+1 in substring is in case of extra space at the end
+        //-1 is because 1 based index is used in command
+        int indexToUnmark = Integer
+                .parseInt(command
+                        .substring(indexTaskName, indexTaskName + 1)) - 1;
+        return indexToUnmark;
+    }
+
+    /**
+     * Parses a command that seeks to add a ToDo o to {@link seedu.duke.task.TaskList}.
+     * @param command is a string with a command keyword "todo"
+     * @return {@link ToDo} to add to {@link seedu.duke.task.TaskList}
+     * @throws DukeException if an incomplete command is given from user
+     */
+    Task parseForTodo(String command) throws DukeException {
+        int indexTaskName = 5;
+        if (command.length() <= 5) {
+            throw new IncompleteCommandException();
+        }
+        String taskName = command.substring(indexTaskName);
+        return new ToDo(taskName);
+    }
+
+    /**
+     * Creates a {@link java.time.LocalDateTime} from a String command detailing the date.
+     * @param dateStr is a string detailing the date in yyyy-MM-dd HH:mm format
+     * @return {@link LocalDateTime} object to create {@link Deadline} and {@link Event}
+     * @throws DukeException if wrongly formatted date or no date is given from user.
+     */
+    LocalDateTime getLocalDateTimeFromDate(String dateStr) throws DukeException {
+        try {
+            TemporalAccessor ta = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm").parse(dateStr);
+            LocalDateTime date = LocalDateTime.from(ta);
+            return date;
+        } catch (DateTimeParseException e) {
+            return null;
+        }
+    }
+
+    /**
+     * Parses a command that seeks to add a {@link Deadline} to {@link seedu.duke.task.TaskList}.
+     * @param command is a string with a command keyword "deadline" from user
+     * @return {@link Deadline} to add to {@link seedu.duke.task.TaskList}
+     * @throws DukeException if an incomplete command is given from user
+     */
+    Task parseForDeadline(String command) throws DukeException {
+        int indexTaskName= 9;
```

Comment:
Perhaps consider leaving a comment to explain why 9, since it'll help another developer understand!

Reply:
Thank you for pointing this out! I have fixed the magic numbers in Parser.java and will take note of similar codes in other files  :+1: